tests/mesh/write_vec_and_scalar.C

```diff
@@ -0,0 +1,131 @@
+
+#include "mesh/curl_curl_system.h"
+#include "test_comm.h"
+
+// THE CPPUNIT_TEST_SUITE_END macro expands to code that involves
+// std::auto_ptr, which in turn produces -Wdeprecated-declarations
+// warnings.  These can be ignored in GCC as long as we wrap the
+// offending code in appropriate pragmas.  We can't get away with a
+// single ignore_warnings.h inclusion at the beginning of this file,
+// since the libmesh headers pull in a restore_warnings.h at some
+// point.  We also don't bother restoring warnings at the end of this
+// file since it's not a header.
+#include <libmesh/ignore_warnings.h>
+
+// Bring in everything from the libMesh namespace
+using namespace libMesh;
+
+class WriteVecAndScalar : public CppUnit::TestCase
+{
+  /**
+   * This test ensures you can write both vector and scalar variables
+   */
+public:
+  CPPUNIT_TEST_SUITE(WriteVecAndScalar);
+
+  CPPUNIT_TEST(testWrite);
+
+  CPPUNIT_TEST_SUITE_END();
+
+  void testWrite()
+  {
+    Mesh mesh(*TestCommWorld);
+
+    MeshTools::Generation::build_square(
+        mesh, 2, 2, -1., 1., -1., 1., Utility::string_to_enum<ElemType>("TRI6"));
+
+    // Create an equation systems object.
+    EquationSystems equation_systems(mesh);
+
+    // Declare the system "Navier-Stokes" and its variables.
+    CurlCurlSystem & system = equation_systems.add_system<CurlCurlSystem>("CurlCurl");
+
+    // This example only implements the steady-state problem
+    system.time_solver = UniquePtr<TimeSolver>(new SteadySolver(system));
+
+    // Initialize the system
+    equation_systems.init();
+
+    // And the nonlinear solver options
+    DiffSolver & solver = *(system.time_solver->diff_solver().get());
+    solver.quiet = true;
+    solver.verbose = !solver.quiet;
+    solver.max_nonlinear_iterations = 15;
+    solver.relative_step_tolerance = 1.e-3;
+    solver.relative_residual_tolerance = 1.0e-13;
+    solver.absolute_residual_tolerance = 0.0;
+
+    // And the linear solver options
+    solver.max_linear_iterations = 50000;
+    solver.initial_linear_tolerance = 1.e-10;
+
+    system.solve();
+
+#ifdef LIBMESH_HAVE_EXODUS_API
+
+    // We write the file in the ExodusII format.
+    ExodusII_IO(mesh).write_equation_systems("out.e", equation_systems);
+
+    // Now read it back in
+    Mesh read_mesh(*TestCommWorld);
+    ExodusII_IO exio(read_mesh);
+    exio.read("out.e");
+    read_mesh.prepare_for_use();
+    EquationSystems es2(read_mesh);
+    ExplicitSystem & sys2 = es2.add_system<ExplicitSystem>("Test");
+
+    const std::vector<std::string> & nodal_vars(exio.get_nodal_var_names());
+
+    for (const auto & var_name : nodal_vars)
+      sys2.add_variable(var_name, SECOND, LAGRANGE);
+
+    es2.init();
+
+    for (const auto & var_name : nodal_vars)
+      exio.copy_nodal_solution(sys2, var_name, var_name, 1);
+
+    std::vector<Real> gold_vector(
+        {-0.020949, -0.020949, 0., 0.009495,  0.183852,  0.,       -0.010570, -0.010570, 0.250000,
```

Comment:
If you end up just writing pre-determined values to the solution vector we can shoot for a tighter tolerance on the checks as well.